File: Online_BookStore_Frontend/src/app/Services/book.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { baseurl } from '../baseURL';
import { Book } from '../book';

@Injectable({
  providedIn: 'root'
})
export class BookService {

  constructor(private httpClient : HttpClient) { }

  public saveBook(book : Book){
    return this.httpClient.post(`${baseurl}/api/book/`,book);
  }

  //get all books
  public getAllBooks(){
    return this.httpClient.get(`${baseurl}/api/book/`);
  }

  public getBookById(bId:number){
    return this.httpClient.get(`${baseurl}/api/book/${bId}`);
  }

  //getBooksByCategory witout that book (similar books display)
  public getBooksByCategory(bCategory:string,bId:number){
    return this.httpClient.get(`${baseurl}/api/book/getBooksByCategory/${bCategory}/${bId}`);
  }

  public findBooksByCategory(bCategory:string){
    return this.httpClient.get(`${baseurl}/api/book/findBooksByCategory/${bCategory}`);
  }

  public findBooksByStatus(bStatus:string){
    return this.httpClient.get(`${baseurl}/api/book/findBooksByStatus/${bStatus}`);
  }

  public updateBook(book : Book,bId:number){
    return this.httpClient.put(`${baseurl}/api/book/${bId}`,book);
  }

  public deleteBook(bId:number){
    return this.httpClient.delete(`${baseurl}/api/book/${bId}`);
  }

}

File: SpringREST-FinalProject/src/main/java/com/example/demo/service/BookService.java
package com.example.demo.service;

import java.util.List;

import com.example.demo.entity.Book;
	
public interface BookService {
	public Book saveBook(Book book);
	
	public List<Book> getAllBooks();
	
	public Book getBookById(int id);
	
	public Book updateBook(int id,Book book);
	
	public List<Book> deleteBook(int id);
	
	public int getTotalBookCountByPublisher(int pId);
	
	public List<Book> getBooksByPublisher(int pId);
	
	//find book by category(similar books display)
	public List<Book> findBooksBybCategory(String bCategory,int bId);
	
	public List<Book> findBybCategory(String bCatogory);
	
	public List<Book> findBybStatus(String bStatus);

	
	//update book sold count
	public Book updateBookSoldCount(int bId,int booksSold);
	
	//update book status
	public Book updateBookStatus(int bId,String bStatus);
	
	//search book
	public List<Book> searchBook(String bTitle);
	
	//get books sold by bId
	public int getBooksSoldByBId(int bId);
	
	public int getBooksSoldByPId(int pId);

}
